packages/library/src/commonMain/kotlin/io/realm/RealmConfiguration.kt

```diff
@@ -17,58 +17,154 @@
+    public constructor(path: String? = null, name: String = Realm.DEFAULT_FILE_NAME, schema: Set<KClass<out RealmObject>>) :
+        this(path, name, mapOf())
 
+    internal constructor(path: String? = null, name: String = Realm.DEFAULT_FILE_NAME, schema: Map<KClass<out RealmObject>, RealmObjectCompanion>) :
+        this(
+            schema,
+            path,
+            name,
+            schema.keys,
+            LogConfiguration(LogLevel.WARN, listOf(PlatformHelper.createDefaultSystemLogger(Realm.DEFAULT_LOG_TAG)))
+        )
+
+    /**
+     * Used to create a [RealmConfiguration]. For common use cases, a [RealmConfiguration] can be created directly
+     * using the [RealmConfiguration] constructor.
+     */
     class Builder(
         var path: String? = null, // Full path for Realm (directory + name)
-        var name: String = "default", // Optional Realm name (default is 'default')
-        vararg var schema: KClass<out RealmObject>
+        var name: String = Realm.DEFAULT_FILE_NAME, // Optional Realm name (default is 'default')
+        var schema: Set<KClass<out RealmObject>> = setOf()
     ) {
+
+        private var logLevel: LogLevel = LogLevel.WARN
+        private var removeSystemLogger: Boolean = false
+        private var userLoggers: List<RealmLogger> = listOf()
+
         fun path(path: String) = apply { this.path = path }
         fun name(name: String) = apply { this.name = name }
-        fun schema(vararg classes: KClass<out RealmObject>) = apply { this.schema = classes }
+        fun schema(classes: Set<KClass<out RealmObject>>) = apply { this.schema = classes }
+        fun schema(vararg classes: KClass<out RealmObject>) = apply { this.schema = setOf(*classes) }
+
+        /**
+         * Configure how Realm will report log events.
+         *
+         * @param level all events at this level or higher will be reported.
+         * @param customLoggers any custom loggers to send log events to. A default system logger is
+         * installed by default that will redirect to the common logging framework on the platform, i.e.
+         * LogCat on Android and NSLog on iOS.
+         */
+        fun log(level: LogLevel = LogLevel.WARN, customLoggers: List<RealmLogger> = emptyList()) = apply {
+            this.logLevel = level
+            this.userLoggers = customLoggers
+        }
+
+        /**
+         * TODO Evaluate if this should be part of the public API. For now keep it internal.
+         *
+         * Removes the default system logger from being installed. If no custom loggers have
+         * been configured, not log events will be reported, regardless of the configured
+         * log level.
+         *
+         * @see [RealmConfiguration.Builder.log]
+         */
+        internal fun removeSystemLogger() = apply { this.removeSystemLogger = true }
 
         fun build(): RealmConfiguration {
             @Suppress("SpreadOperator")
```

Comment:
Not needed anymore, but maybe at https://github.com/realm/realm-kotlin/pull/171/files#diff-2ee4dd35764b2be7198c9da2591563685604ccc9a3580256919303340bdabeb7R118